lnrpc/routerrpc/router_server.go

```diff
@@ -737,19 +741,56 @@ func (s *Server) trackPayment(identifier lntypes.Hash,
 	router := s.cfg.RouterBackend
 
 	// Subscribe to the outcome of this payment.
-	subscription, err := router.Tower.SubscribePayment(
-		identifier,
-	)
+	subscription, err := router.Tower.SubscribePayment(identifier)
+
 	switch {
 	case err == channeldb.ErrPaymentNotInitiated:
 		return status.Error(codes.NotFound, err.Error())
 	case err != nil:
 		return err
 	}
+
+	// Stream updates to the client.
+	err = s.trackPaymentStream(
+		stream.Context(), subscription, noInflightUpdates, stream.Send,
+	)
+
+	if errors.Is(err, context.Canceled) {
+		log.Debugf("Payment stream %v canceled", identifier)
+	}
+
+	return err
+}
+
+// TrackPayments returns a stream of payment state updates.
+func (s *Server) TrackPayments(request *TrackPaymentsRequest,
+	stream Router_TrackPaymentsServer) error {
```

Comment:
Can we use `Router_TrackPaymentsV2Server` here? `Router_TrackPaymentsServer` is deprecated.

Reply:
Do you mean rename the newly created method `TrackPayments` to `TrackPaymentsV2`? I think you may be confused with `TrackPayment` (without the `s`)?

Reply:
hah yeah my bad! Didn't see the `s`.